tests/lib/test_openstack_identity.py

```diff
@@ -197,3 +197,25 @@ def test_find_user_returns_result(self):
         )
         expected = self.identity_api.find_user.return_value
         assert returned == expected
+
+    @raises(MissingMandatoryParamError)
+    def test_find_user_all_domains_missing_identifier(self):
+        """
+        Checks a missing user identifier will raise correctly
+        """
+        self.instance.find_user_all_domains(NonCallableMock(), " \t")
+
+    def test_find_user_all_domains_returns_result(self):
+        """
+        Checks that find_user_all_domains returns the correct result
+        """
+        cloud, user = NonCallableMock(), NonCallableMock()
+        returned = self.instance.find_user_all_domains(cloud, user)
+
+        self.mocked_connection.assert_called_once_with(cloud)
+        self.identity_api.find_user.assert_called_once_with(
```

Comment:
Do we know what will happen if a duplicate record is found? Will it throw a `MultipleRecords` exception or return both?

Reply:
Speaking to Alex & Jacob sounds like it would likely just error as keystone would return a 501 error. Not sure how the API would react to that, but I would think its likely it would just crash there before sending any emails.

Reply:
Lets apply YAGNI to this then